File: data/src/main/java/com/example/data/database/NasaDao.java
package com.example.data.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.data.model.APODRoom;

import java.util.List;

@Dao
public interface NasaDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAstronomyPicture(APODRoom apodJson);

    @Query("select * from APODRoom where date = :date")
    APODRoom getAstronomyPicture(String date);

    @Query("select * from APODRoom")
    List<APODRoom> getAll();
}

File: data/src/main/java/com/example/data/database/NasaDao.kt
package com.example.data.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.data.model.APODRoom

@Dao
interface NasaDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAstronomyPicture(apodRoom: APODRoom)

    @Query("select * from APODRoom where date = :date")
    suspend fun getAstronomyPicture(date: String): APODRoom?

    @Query("select * from APODRoom")
    suspend fun getAll(): List<APODRoom>
}